src/main/java/com/linecorp/armeria/server/docs/ThriftDocString.java

```diff
@@ -0,0 +1,172 @@
+import java.io.InputStream;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import javax.xml.stream.XMLStreamReader;
+
+import org.reflections.Reflections;
+import org.reflections.scanners.ResourcesScanner;
+import org.reflections.util.ClasspathHelper;
+import org.reflections.util.ConfigurationBuilder;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Strings;
+import com.google.common.collect.Collections2;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableMap.Builder;
+
+/**
+ * {@link ThriftDocString} is an DocString extractor for Thrift IDL JSON.
+ * If you want use DocString in {@link DocService}, generate YOUR.json using YOUR.thrift and
+ * latest Thrift compiler with --gen json option. And, place YOUR.json file to
+ * resources/&#42;&#42;&#47;thriftjson&#47;&#42;&#42;&#47;YOUR.json of package paths.
+ */
+public final class ThriftDocString {
+
+    private static final Pattern FILENAME_FILTER = Pattern.compile(".*json");
+
+    private static final Pattern FULLPATH_FILTER = Pattern.compile(".*/thriftjson/.*\\.json");
+
+    private static final TypeReference JSON_VALUE_TYPE = new TypeReference<HashMap<String, Object>>() {};
+
+    private static final String FQCN_DELIM = ".";
+
+    private static final String DELIM = "#";
+
+    private static Map<ClassLoader, Map<String, String>> cached = new ConcurrentHashMap<>();
+
+    private ThriftDocString() {
+    }
+
+    /**
+     * Extracts all DocStrings from all Thrift IDL JSON Resources.
+     * @return a map with key is FQCN and value is document string.
+     */
+    static Map<String, String> getAllDocStrings(ClassLoader classLoader) {
+        return cached.computeIfAbsent(classLoader, loader -> parseDocStrings(loader, getAllThriftJsons(loader)));
+    }
+
+    /**
+     * Parses DocStrings from input Thrift IDL JSON Resources.
+     * @return a map with key is FQCN and value is document string.
+     */
+    static Map<String, String> parseDocStrings(ClassLoader classLoader, Collection<String> jsonPaths) {
+        final ImmutableMap.Builder<String, String> docStrings = ImmutableMap.builder();
+        for (String jsonPath : jsonPaths) {
+            final Map<String, String> values = getDocStringsFromJsonResource(classLoader, jsonPath);
+            if (values != null) {
+                docStrings.putAll(values);
+            }
+        }
+        return docStrings.build();
+    }
+
+    static Collection<String> getAllThriftJsons(ClassLoader classLoader) {
+        final Reflections reflections = new Reflections(new ConfigurationBuilder()
+                                                                .setUrls(ClasspathHelper.forPackage(""))
```

Comment:
I tried this, but it's not suitable for single package. I looks like we cannot avoid to scan full classpaths.

### 1. Using a specific package in unittests.

codes:
```
System.out.println(ClasspathHelper.forPackage("com.linecorp.armeria.server.docs.thriftjson"));
```
output:
```
[file:/Users/username/working-directory/armeria/target/test-classes/]
```

### 2. Using it without specific package in unittests.
codes:
```
System.out.println(ClasspathHelper.forPackage(""));
```
output:
```
[file:/Users/username/working-directory/armeria/target/test-classes/, file:/Users/username/working-directory/armeria/target/classes/]
```